src/crawlers/crawler_extension.js

```diff
@@ -3,8 +3,8 @@
  * Abstract class with pre-defined method to connect to the Crawlers class by the "use" crawler method.
  */
 export default class CrawlerExtension {
-    constructor(options) {
-        this.name = options.name;
+    constructor() {
+        this.name = this.constructor.name;
```

Comment:
Do we need the name in the extension itself? Or is it just for the logging in `use`? We could use the `ext.constructor.name` there and get rid of the constructor here.

Reply:
I have it in the extension itself for errors and logging.

Reply:
The new version of `log` that's now in master, but not yet released in latest, allows you to create child logger instances. Maybe we could use that and do `log.child({ prefix: this.constructor.name })` ? We could also "use" the Crawler's logger as parent, in the `use` function, to get nested prefixes e.g. `CheerioCrawler:CloudflareUnblocker`

And while doing the logging refactoring, we decided that we will not prefix log messages, but rather create specific error instances. But this exists only as an issue yet #646

Reply:
As discussed with @mnmkng every extension should have its own logging since it is initialized outside of the Crawler.